.github/workflows/e2e-test-tf-module.yml

```
@@ -0,0 +1,191 @@
+          last_three="${run_id: -3}"
+          echo "prefix=e2e-${last_three}-${{ github.run_attempt }}" | tee -a "$GITHUB_OUTPUT"
+
+      - name: Create Terraform variable input file
+        if: inputs.cloudProvider == 'aws'
+        working-directory: ./terraform-module/aws-constellation
+        run: |
+          cat > terraform.tfvars <<EOF
+          name = "${{ steps.create-prefix.outputs.prefix }}"
+          image = "${{ inputs.image }}"
+          zone = "${{ inputs.regionZone }}"
+          name_prefix = "${{ steps.create-prefix.outputs.prefix }}"
+          node_groups = {
+            control_plane_default = {
+              role          = "control-plane"
+              zone          = "${{ inputs.regionZone }}"
+              instance_type = "m6a.xlarge"
+              disk_size     = 30
+              disk_type     = "gp3"
+              initial_count = 2
+            },
+            worker_default = {
+              role          = "worker"
+              zone          = "${{ inputs.regionZone }}"
+              instance_type = "m6a.xlarge"
+              disk_size     = 30
+              disk_type     = "gp3"
+              initial_count = 2
+            }
+          }
+          EOF
+          cat terraform.tfvars
+
+      - name: Install dependencies (Terraform)
+        run: |
+          sudo apt update && sudo apt install gpg
+          wget -O- https://apt.releases.hashicorp.com/gpg | sudo gpg --dearmor -o /usr/share/keyrings/hashicorp-archive-keyring.gpg
+          gpg --no-default-keyring --keyring /usr/share/keyrings/hashicorp-archive-keyring.gpg --fingerprint
+          echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/hashicorp-archive-keyring.gpg] https://apt.releases.hashicorp.com $(lsb_release -cs) main" | sudo tee /etc/apt/sources.list.d/hashicorp.list
+          sudo apt update
+          sudo apt install terraform=1.4.4-*
+
+      - name: Setup bazel
+        uses: ./.github/actions/setup_bazel_nix
+        with:
+          useCache: "true"
+          buildBuddyApiKey: ${{ secrets.BUILDBUDDY_ORG_API_KEY }}
+
+      - name: Log in to the Container registry
+        uses: ./.github/actions/container_registry_login
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build CLI
+        if: inputs.cliVersion == ''
+        uses: ./.github/actions/build_cli
+        with:
+          outputPath: "constellation"
+          enterpriseCLI: true
+          push: true
+
+      - name: Download CLI
+        if: inputs.cliVersion != ''
+        shell: bash
+        run: |
+          mkdir build
+          curl -fsSL -o constellation https://github.com/edgelesssys/constellation/releases/download/${{ inputs.cliVersion }}/constellation-linux-amd64
+          chmod u+x constellation
+          PATH_ADD=$(pwd)
+          export PATH="$PATH:$PATH_ADD"
+          constellation version
```

```suggestion
          chmod u+x ./constellation
          PATH_ADD=$(pwd)
          export PATH="$PATH:$PATH_ADD"
          ./constellation version
```
ensure local file is used in case of collisions with an already-installed CLI